sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -575,16 +577,102 @@ func cleanUp(containerdSocket string, namespace string, containerID string) erro
 	return nil
 }
 
+// parseImageURISpecialRegions mimics the parsing in ecr.ParseImageURI but
+// constructs the canonical ECR references while skipping certain checks.
+// We only do this for special regions that are not yet supported by the aws-go-sdk.
+// Referenced source: https://github.com/awslabs/amazon-ecr-containerd-resolver/blob/a5058cf091f4fc573813a032db37a9820952f1f9/ecr/ref.go#L70-L71
+func parseImageURISpecialRegions(input string) (ecr.ECRSpec, error) {
+	ecrRefPrefixMapping := map[string]string{
+		"il-central-1": "ecr.aws/arn:aws:ecr:il-central-1:",
+	}
+	// Matching on account, region
+	matches := ecrRegex.FindStringSubmatch(input)
+	if len(matches) < 3 {
+		return ecr.ECRSpec{}, fmt.Errorf("invalid image URI: %s", input)
+	}
+	account := matches[1]
+	region := matches[2]
+
+	// Need to include the full repository path and the imageID (e.g. /eks/image-name:tag)
+	tokens := strings.SplitN(input, "/", 2)
+	if len(tokens) != 2 {
+		return ecr.ECRSpec{}, fmt.Errorf("invalid image URI: %s", input)
+	}
+	fullRepoPath := tokens[len(tokens)-1]
+	// Run simple checks on the provided repository.
+	switch {
+	case
+		// Must not be empty
+		fullRepoPath == "",
+		// Must not have a partial/unsupplied label
+		strings.HasSuffix(fullRepoPath, ":"),
+		// Must not have a partial/unsupplied digest specifier
+		strings.HasSuffix(fullRepoPath, "@"):
+		return ecr.ECRSpec{}, errors.New("incomplete reference provided")
+	}
+
+	// Parse out image reference's to validate.
+	ref, err := reference.Parse("repository/" + fullRepoPath)
+	if err != nil {
+		return ecr.ECRSpec{}, err
+	}
+	// If the digest is provided, check that it is valid.
+	if ref.Digest() != "" {
+		err := ref.Digest().Validate()
+		// Digest may not be supported by the client despite it passing against
+		// a rudimentary check. The error is different in the passing case, so
+		// that's considered a passing check for unavailable digesters.
+		//
+		// https://github.com/opencontainers/go-digest/blob/ea51bea511f75cfa3ef6098cc253c5c3609b037a/digest.go#L110-L115
+		if err != nil && err != digest.ErrDigestUnsupported {
+			return ecr.ECRSpec{}, fmt.Errorf("%v: %w", "invalid image URI", err)
+		}
+	}
```

Comment:
Do we need these two checks? The first (`reference.Parse()`) duplicates one in `ParseARN`, and it's not clear why we need to check the digest here when `ParseARN` doesn't have that check.

```suggestion
```